Pull Request : ISD Group H

HellOrWorld opened the pull request:
FUNCTIONALITIES IMPLEMENTED

1. Task 1: Image Histogram

> Web page can take any image from the imagenet_subset and return an Histogram using Javascript.

> Additional Functionalities: Download results as .txt and graph buttons (implemented in the same way as Task 2)

2. Task 2: Dowload Image Classifiactions Results/Graph
Buttons

> Web page can download onto the users computer both the image classification graph and the classifications results as JSON.


ISSUES RESOLVED:

- Fixed the concurrency issue in the dowload results buttons (task 1 & 2)
-Deleted all the versioned files that served no purpose (were old attemps at making an histogram using matplotlip)


Made by Group H (Erasmus):

Gabriel LA ROSA--RENCUREL (HellOrWorld)
Theo DE SAUVAGE (desaah)

HellOrWorld committed:
Update on Readme.md concerning the execution

HellOrWorld committed:
Histogram html display

Commit:
First

Commit:
second

HellOrWorld committed:
Task 2: Download Results & Plot buttons on Image Classification Page

HellOrWorld committed:
Task 2: Download Results & Plot buttons on Image Classification Page

HellOrWorld committed:
Task 1: Image Histogram calculation and display

HellOrWorld committed:
Final Push: Comments and Optimisation as well as the end of Task 1

asotgiu commented on app/static/histograms/213aa9ff-4edd-4963-b765-81a4dbd24dbf.png:
Why is this file versioned?

asotgiu commented on app/static/histograms/4c66aaa3-7d94-4d73-8ccc-993679e5fc50.png:
Why is this file versioned?

asotgiu commented on app/static/histograms/7b0cba20-099b-4345-be82-08bdb26de22e.png:
Why is this file versioned?

asotgiu commented on app/static/histograms/89137ba7-d63c-4e35-b583-2345d7c99331.png:
Why is this file versioned?

asotgiu commented on main.py:
```diff
@@ -37,20 +38,33 @@ def home(request: Request):
+    # We classify the image using the model
     classification_scores = classify_image(model_id=model_id, img_id=image_id)
+
+    # We save the classification scores to the classification_scores.json file
+    with open('app/static/classification_scores.json', 'w') as f:
+        json.dump(classification_scores, f)
```
What happens if multiple users ask for a classification job? The file would be overwritten. Isn't it best to return the results directly? Or, if you reuse them later for other things, another way to do it would be to store the results with a unique name and clean up after the response is returned to the user (or after a while anyway).

asotgiu commented on main.py:
```diff
@@ -59,3 +73,55 @@ async def request_classification(request: Request):
+    # We calculate the histogram scores for the image
+    histogram_scores = calculate_histogram(image_id)
+
+    # We save the histogram scores to the histogram_scores.txt file
+    with open('app/static/histogram_scores.txt', 'w') as f:
+        f.write(str(histogram_scores))
```
What happens if multiple users ask for an image histogram? The file would be overwritten. Isn't it best to return the results directly? Or, if you reuse them later for other things, another way to do it would be to store the results with a unique name and clean up after the response is returned to the user (or after a while anyway).

asotgiu commented on histogram:
What is this file, and why it is versioned?

asotgiu reviewed (changes requested):
I reviewed the code, but some issues must be solved before approving and merging the Pull Request. Please fix them and request a new review when it's done. There's no need to open a new PR, it's enough to push the new commits to the origin repository (yours) of the PR. In that way, I will see only the new changes and I won't have to review the full code again.
- fix concurrency problems;
- remove files that should not be versioned;
- add a changelog in the Pull Request description, where you describe the work you have done and link there the solved issues.

Thank you!

HellOrWorld committed:
Final Push: Comments and Optimisation as well as the end of Task 1

HellOrWorld committed:
Revisions and corrections done

HellOrWorld committed:
Revisions and corrections done

HellOrWorld committed:
Merge remote-tracking branch 'origin/main'

HellOrWorld commented:
Good morning @asotgiu,

Sorry to disturb you.

We've fixed all the issues and and made the adjusted corrections to our code.
We will let you review them.

FUNCTIONALITIES IMPLEMENTED

-Task 1: Image Histogram

Web page can take any image from the imagenet_subset and return an Histogram using Javascript.

Additional Functionalities: Download results as .txt and
graph buttons (implemented in the same way as Task 2)

-Task 2: Dowload Image Classifiactions Results/Graph
Buttons

Web page can download onto the users computer both the image classification grap and the classifications results as JSON.


ISSUES RESOLVED

- Fixed the concurrency issue in the dowload results buttons (task 1 & 2)
-Deleted all the versioned files that served no purpose (were old attemps at making an histogram using matplotlip)

NOTE:

Everything should be fixed and there should not be any other problems.

Don't hesitate to let us hnow if there are any other issues to fix or to functionalities to implment.

Gabriel LA ROSA--RENCUREL (HellOrWorld)
Theo DE SAUVAGE (desaah)

asotgiu reviewed (approved):
I reviewed again the code and it looks good now. Well done! I will approve and merge the PR.
